Modules/Scripted/DICOMLib/DICOMProcesses.py

```diff
@@ -436,21 +443,50 @@ def send(self):
         if not self.progressCallback(f"Sent {file} to {self.destinationUrl.host()}:{self.destinationUrl.port()}"):
           raise UserWarning("Sending was cancelled, upload is incomplete.")
 
-  def start(self,file):
+  def start(self, file, config=None, config_profile='Default', retry=False):
```

Comment:
Adding a Bool flag and introducing recursion would make things complicated and fragile. Instead, it would be much cleaner if you factored out the parts that you want to re-run into a separate method and call that method with various parameters from the `start` method.

Reply:
The method has been refactored to remove the recursion. The transfer logic, method that waits until the tx ends, and process status have been added to a helper method `dicomSend` that is called twice: once with the default configuration and a second time with the alternative configuration.